calyx-opt/src/passes/static_promotion.rs

```diff
@@ -523,31 +642,86 @@ impl Visitor for StaticPromotion {
     fn finish_seq(
         &mut self,
         s: &mut ir::Seq,
-        _comp: &mut ir::Component,
-        _sigs: &LibrarySignatures,
+        comp: &mut ir::Component,
+        sigs: &LibrarySignatures,
         _comps: &[ir::Component],
     ) -> VisResult {
-        if let Some(sseq) = s.make_static() {
-            return Ok(Action::change(ir::Control::Static(
-                ir::StaticControl::Seq(sseq),
-            )));
+        let mut builder = ir::Builder::new(comp, sigs);
+        let mut new_stmts: Vec<ir::Control> = Vec::new();
+        let mut static_vec: Vec<ir::Control> = Vec::new();
+        let mut promote_control = true;
+        for stmt in std::mem::take(&mut s.stmts) {
+            if stmt.is_static()
+                || stmt.has_attribute(ir::NumAttr::PromoteStatic)
+            {
+                static_vec.push(stmt);
+            } else {
+                promote_control = false;
+                match static_vec.len().cmp(&1) {
```

Comment:
Maybe worth changing this in a future PR to something like `//static_vec contains a single enable`: since what if `static_vec` contains a `@promote_static par` or something that would be helpful to promote? 
Also, either way, adding a comment might be helpful for future reference so that we understand the code when we're making better heuristics.